### PR TITLE
FDR: Controller R/W instrumentation

### PR DESCRIPTION
**Pull request by kp6**
isolated FDR from other changes

### CODE REVIEW

File: backend/remote/remote.go [hunk 1]
```diff
@@ -168,29 +169,38 @@ func (rf *Factory) Create(address string) (types.Backend, error) {
...
+	go r.monitorPing(rpc, dataAddress /*use replica data addr for ping tracking*/)
 
 	return r, nil
 }
 
-func (r *Remote) monitorPing(client *rpc.Client) error {
+func (r *Remote) monitorPing(client *rpc.Client, targetAddr string) error {
 	ticker := time.NewTicker(pingInveral)
 	defer ticker.Stop()
+	{
```
Comment (ibuildthecloud) [inline backend/remote/remote.go]: what's the code block for?

File: backend/remote/remote.go [hunk 2]
```diff
@@ -168,29 +169,38 @@ func (rf *Factory) Create(address string) (types.Backend, error) {
...
+		// resolve target addr to match tracking of r/w ops
+		tcpAddr, err := net.ResolveTCPAddr("", targetAddr)
```
Comment (ibuildthecloud) [inline backend/remote/remote.go]: Why resolve the backend?

File: controller/rest/router.go [hunk 3]
```diff
@@ -32,5 +32,8 @@ func NewRouter(s *Server) *mux.Router {
 	router.Methods("DELETE").Path("/v1/replicas/{id}").Handler(f(schemas, s.DeleteReplica))
 	router.Methods("PUT").Path("/v1/replicas/{id}").Handler(f(schemas, s.UpdateReplica))
 
+	// Stats
+	router.Methods("POST").Path("/v1/stats").Handler(f(schemas, s.ListStats))
```
Comment (ibuildthecloud) [inline controller/rest/router.go]: GET is more fitting here.

File: rpc/client.go [hunk 4]
```diff
@@ -126,13 +127,20 @@ func (c *Client) nextSeq() uint32 {
...
 func (c *Client) replyError(req *Message) {
+	stats.RemovePendingOp(req.ID, false)
 	delete(c.messages, req.Seq)
 	req.Type = TypeError
 	req.Data = []byte(c.err.Error())
 	req.Complete <- struct{}{}
 }
 
 func (c *Client) handleRequest(req *Message) {
+	switch req.Type {
```
Comment (ibuildthecloud) [inline rpc/client.go]: The RPC layer is all async, but higher level code like replica and control are all synchronous.  It makes more sense to annotate the code for this ops tracking at a higher level.  Then the code will be straight forward, `InsertPendingOp(); ReadAt(); RemovePendingOp()`

File: app/stats.go [hunk 5]
```diff
@@ -0,0 +1,26 @@
+package app
+
+import (
+	"github.com/Sirupsen/logrus"
+	"github.com/codegangsta/cli"
+)
+
+func LsStats() cli.Command {
+	return cli.Command{
+		Name:      "ls-stats",
```
Comment (ibuildthecloud) [inline app/stats.go]: What about journal, log, or telemetry.  Seems like we will add stats later and it will be key/values, not this log data.

**Comment by kp6** [comment]
@ibuildthecloud  rebased

**Pull request merged**